### ISSUE TITLE
I got this unhandled exception when I have attached the  TaskScheduler.UnobservedTaskException:

### ISSUE THREAD

**Issue by richardschneider** [richardschneider/net-mdns#83]
I got this unhandled exception when I have attached the  TaskScheduler.UnobservedTaskException:

 

```2019-10-16 16:17:02.429 +03:00 (14732:0002) [ERR] AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread.

   SocketException: A request to send or receive data was disallowed because the socket had already been shut down in that direction with a previous shutdown call
   at System.Net.Sockets.Socket.DoBeginReceiveFrom(Byte[], Int32, Int32, System.Net.Sockets.SocketFlags, System.Net.EndPoint, System.Net.SocketAddress, System.Net.Sockets.OverlappedAsyncResult);System.dll;100671645;212
   at System.Net.Sockets.Socket.BeginReceiveFrom(Byte[], Int32, Int32, System.Net.Sockets.SocketFlags, System.Net.EndPoint ByRef, System.AsyncCallback, System.Object);System.dll;100671644;270
   at System.Net.Sockets.UdpClient.BeginReceive(System.AsyncCallback, System.Object);System.dll;100671973;42
   at System.Threading.Tasks.TaskFactory`1.FromAsyncImpl(System.Func`3[System.AsyncCallback,System.Object,System.IAsyncResult], System.Func`2[System.IAsyncResult,TResult], System.Action`1[System.IAsyncResult], System.Object, System.Threading.Tasks.TaskCreationOptions);mscorlib.dll;100679367;304
   at System.Net.Sockets.UdpClient.ReceiveAsync();System.dll;100671984;17
   at Makaretu.Dns.MulticastClient+<>c__DisplayClass13_0+<<Listen>b__0>d.MoveNext();Makaretu.Dns.Multicast.dll;100663321;23 
```

@richardschneider does #81 also fix this. If not maybe the receive code should catch all exceptions?

_Originally posted by @Luai-m in https://github.com/richardschneider/net-mdns/issues/79#issuecomment-543015631_

**Comment by richardschneider** [comment]
@Luai-m opened a new issue

**Comment by richardschneider** [comment]
@luai-m, you are correct; changing the `Listen`  method to catch any exception.

**Comment by Luai-m** [comment]
When will the fixed be released?

**Comment by richardschneider** [comment]
v0.26.0 released
